cloud/WFLY-11824_wildfly_operator_kubernetes.adoc

```diff
@@ -0,0 +1,147 @@
+= WildFly Operator for Kubernetes/OpenShift
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Managing an application running on WildFly in a cloud environment can be a complex task. Things like ensuring containers have access to necessary resources, scaling up and scale down efficiently, integrating with outside services and handling rollout of new versions are all complex tasks that require operation knowledge. An https://coreos.com/blog/introducing-operators.html[Operator] is a specialized form of Kubernetes / OpenShift controller that is meant to encode that kind of operation knowledge in software, allowing users to more easily manage their cloud applications. The goal of this feature is to develop and Operator for managing WildFly-based applications.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11824[WFLY-11824]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1149[EAP7-1149] - Tech Preview Operator for handling already built EAP/WildFly based images
+* https://issues.jboss.org/browse/CLOUD-2262[CLOUD-2262] - Provide support for transaction recovery for transactions which contain subordinate transactions propagated over JBoss Remoting
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+ * mailto:mchoma@redhat.com[Martin Choma]
+
+=== Testing By
+
+[x] QE
+
+=== Affected Projects or Components
+
+ * https://github.com/wildfly/wildfly[WildFly]
+ * https://github.com/openshift-s2i/s2i-wildfly[WildFly s2i]
+ * https://www.github.com/jboss-dockerfiles/wildfly[WildFly Docker image]
+
+
+=== Other Interested Projects
+
+ * https://github.com/jboss-dockerfiles/wildfly[WildFly Docker image]
+ * Keycloak / Red Hat SSO
+ * RHPAM
+
+== Requirements
+
+
+=== Hard Requirements
+
+==== Create a Custom Resource Definition (CRD) for WildFly.
+
+Provide a Custom Resource Definition (CRD) for an application based on a WildFly application server.
+
+The CRD will allow the user to:
+
+* Specify the name an application image that incorporates a WildFly server and an deployment that will be deployed by that server.
+** The name can include either a fixed or floating tag.
+** Application images must be built on top of [WildFly S2I](https://github.com/wildfly/wildfly-s2i) starting with WildFly 17.
+** The Operator should be able to manage operatios built with the [legacy WildFly S2I for WildFly 16](https://github.com/openshift-s2i/s2i-wildfly)
+** Application images built on top of older WildFly versions are not supported.
+* Specify the desired number of replicas of that image
+* Specify an optional ConfigMap where the WildFly standalone configuration can be read from.
+* Specify whether the replicas are intended to form a High Availability cluster (default is true).
```

Comment:
Which attribute from https://github.com/wildfly/wildfly-operator/blob/master/doc/apis.adoc is that?